Doc 1395 - audit logs for user actions (#139)
tigergraph/server-docs: modules/security/pages/audit-privileged-actions.adoc

For example:

[tabs]
====
Command::
+
--
[source.wrap,console]
$ awk '/^(E|I|D)@.*/ {p=0}; /^(E|I|D)@.*_ACCESS_CHECK_.*/ {p=1; printf FILENAME ":" FNR ":"}; p { print }' \
~/tigergraph/log/gsql/INFO.*
--
Results::
+
--
[source,text]
/home/tigergraph/tigergraph/log/gsql/INFO.20220914-163940.595:1200:I@20220914 16:44:11.344 tigergraph|127.0.0.1:34994|00000000013 (PermissionUtil.java:371) User Name: tigergraph, _ACCESS_CHECK_RBAC_, Command Type: create query hello ( ) { print hello ; }, Validation Result: SUCCESS, Required RBAC Permission: [], Required Data Access Permission: [], Required Application Access Permission: []
/home/tigergraph/tigergraph/log/gsql/INFO.20220914-163940.595:1532:I@20220914 19:27:34.585 tigergraph|127.0.0.1:41468|00000000013 (PermissionUtil.java:371) User Name: tigergraph, _ACCESS_CHECK_RBAC_, Command Type: /gsql/abortclientsession, Validation Result: SUCCESS, Required RBAC Permission: [], Required Data Access Permission: [], Required Application Access Permission: []

--
====

== Extract logs by user

Run the following command in a bash shell to extract information on all privileged actions on an instance by a particular user.

> Replace `<username>` in the first line with the name of the user whose actions you want to look up:

Reviewer: Might worth mentioning that <username> is also case sensitive here in the example command.

`awk` does have some mechanism for allowing case insensitive regex pattern matching. But it may be beyond the scope of this page's example scope.

--
Results::
+
--
[source,text]
/home/tigergraph/tigergraph/log/gsql/INFO.20220914-163940.595:1200:I@20220914 16:44:11.344 tigergraph|127.0.0.1:34994|00000000013 (PermissionUtil.java:371) User Name: tigergraph, _ACCESS_CHECK_RBAC_, Command Type: create query hello ( ) { print hello ; }, Validation Result: SUCCESS, Required RBAC Permission: [], Required Data Access Permission: [], Required Application Access Permission: []
/home/tigergraph/tigergraph/log/gsql/INFO.20220914-163940.595:1532:I@20220914 19:27:34.585 tigergraph|127.0.0.1:41468|00000000013 (PermissionUtil.java:371) User Name: tigergraph, _ACCESS_CHECK_RBAC_, Command Type: /gsql/abortclientsession, Validation Result: SUCCESS, Required RBAC Permission: [], Required Data Access Permission: [], Required Application Access Permission: []

--
====

== Extract logs by action

You can also look up specific actions that occurred on an instance and investigate when and by whom did those actions occur.

* For GSQL commands, the "Command Type" field of a log details the exact command that was executed.

> * For REST API cals, the "Command Type" field of a log details the endpoint that was called.

Reviewer: typo: calls

Run the following command in a bash shell to extract information on all privileged actions of a particular type on an instance.
Replace `<command>` with the type of action you want to look up.
The content of the command is case-sensitive:

[.wrap,console]
----
awk '/^(E|I|D)@.*/ {p=0}; /^(E|I|D)@.*_ACCESS_CHECK_.*Command Type: <command>.*/ {p=1; printf FILENAME ":" FNR ":"}; \
p { print }' ~/tigergraph/log/gsql/INFO.* <1>
----
<1> This pattern contains the default log location.
If your logs are located at a different path, replace it with your log location.

For example:

[tabs]
====
Command::
+
--

> The following request prints all privileged actions by the user `tigergraph` on this instance:

Reviewer: This one should be request prints `CREATE GRAPH` action by all users.

For example:

[tabs]
====
Command::
+
--
The following request prints all privileged actions by the user `tigergraph` on this instance:

[source.wrap,console]
$ awk '/^(E|I|D)@.*/ {p=0}; /^(E|I|D)@.*User Name: tigergraph.*_ACCESS_CHECK_.*/ {p=1; printf FILENAME ":" FNR ":"}; p { print }' ~/tigergraph/log/gsql/INFO.*
--
Results::
+
--
[source,text]
/home/tigergraph/tigergraph/log/gsql/INFO.20220914-163940.595:1200:I@20220914 16:44:11.344 tigergraph|127.0.0.1:34994|00000000013 (PermissionUtil.java:371) User Name: tigergraph, _ACCESS_CHECK_RBAC_, Command Type: create query hello ( ) { print hello ; }, Validation Result: SUCCESS, Required RBAC Permission: [], Required Data Access Permission: [], Required Application Access Permission: []
/home/tigergraph/tigergraph/log/gsql/INFO.20220914-163940.595:1532:I@20220914 19:27:34.585 tigergraph|127.0.0.1:41468|00000000013 (PermissionUtil.java:371) User Name: tigergraph, _ACCESS_CHECK_RBAC_, Command Type: /gsql/abortclientsession, Validation Result: SUCCESS, Required RBAC Permission: [], Required Data Access Permission: [], Required Application Access Permission: []

--
====

== Extract logs by action

> You can also look up specific actions that occurred on an instance and investigate when and by whom did those actions occur.

Reviewer 2: ```suggestion
You can also look up specific actions that occurred on an instance and investigate when and by whom those actions were performed.
```
